tests/models/test_timestamp.py

```diff
@@ -0,0 +1,76 @@
+# Licensed to the Apache Software Foundation (ASF) under one
+# or more contributor license agreements.  See the NOTICE file
+# distributed with this work for additional information
+# regarding copyright ownership.  The ASF licenses this file
+# to you under the Apache License, Version 2.0 (the
+# "License"); you may not use this file except in compliance
+# with the License.  You may obtain a copy of the License at
+#
+#   http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing,
+# software distributed under the License is distributed on an
+# "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+# KIND, either express or implied.  See the License for the
+# specific language governing permissions and limitations
+# under the License.
+
+import pendulum
+import pytest
+from freezegun import freeze_time
+
+from airflow.models import DAG, Log, TaskInstance
+from airflow.operators.dummy_operator import DummyOperator
+from airflow.utils import timezone
+from airflow.utils.session import provide_session
+from airflow.utils.state import State
+
+
+@pytest.yield_fixture()
+@provide_session
+def clear_db(session=None):
+    print("Clearing before")
+    session.query(Log).delete()
+    session.query(TaskInstance).delete()
+    yield session
+    print("Clearing after")
+    session.query(Log).delete()
+    session.query(TaskInstance).delete()
+    session.commit()
+
+
+def add_log(execdate, session, timezone_override=None):
+    dag = DAG(dag_id='logging',
+              default_args={'start_date': execdate})
+    task = DummyOperator(task_id='dummy', dag=dag, owner='airflow')
+    task_instance = TaskInstance(task=task, execution_date=execdate, state='success')
+    session.merge(task_instance)
+    log = Log(State.RUNNING, task_instance)
+    if timezone_override:
+        log.dttm = log.dttm.astimezone(timezone_override)
+        assert log.dttm.tzinfo.name != 'UTC'
+    session.add(log)
+    session.commit()
+
+
+def test_timestamp_behaviour(clear_db):  # pylint: disable=redefined-outer-name
+    for session in clear_db:
+        execdate = timezone.utcnow()
+        with freeze_time(execdate):
+            current_time = timezone.utcnow()
+            add_log(execdate, session)
+            log_time = session.query(Log).one().dttm
+            assert log_time == current_time
+            assert log_time.tzinfo.name == 'UTC'
+
+
+def test_timestamp_behaviourwith_timezone(clear_db):  # pylint: disable=redefined-outer-name
+    for session in clear_db:
+        execdate = timezone.utcnow()
+        with freeze_time(execdate):
+            current_time = timezone.utcnow()
+            add_log(execdate, session, timezone_override=pendulum.timezone('Europe/Warsaw'))
+            # No matter what timezone we set - we should always get back UTC
+            log_time = session.query(Log).one().dttm
+            assert log_time == current_time
+            assert log_time.tzinfo.name == 'UTC'
```

Comment:
I'd feel a bit safer here if we also tested something about the time -- that the tz's are different _and_ that the same "time" is given for both.

Reply:
Indeed. I also added expunge to make sure that we are not getting the Log object from cache. And now it's much nicer to have all asserts in one place.